src/store/metadata/sparql.py

```diff
@@ -0,0 +1,230 @@
+from typing import Dict, Optional
+
+from rdflib import Graph
+from rdflib.term import Identifier
+
+from custom_logging import logger
+
+
+def construct(
+    query: str, graph: Graph, init_bindings: Optional[Dict[str, Identifier]] = None
```

Comment:
where we have init_bindings we'll want to validate them in some way if possible. If the queries borked because of bad bindings its worth us explicitly being told to save us having to unpick sparql.